tee-worker/local-setup/launch.py

```diff
@@ -44,12 +50,54 @@ def run_worker(config, i: int):
     print(f'Starting worker {i} in background')
     return w.run_in_background(log_file=log, flags=config["flags"], subcommand_flags=config["subcommand_flags"])
 
+# Function to check if a port is open
+def is_port_open(port):
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.bind(('127.0.0.1', int(port)))
+        sock.close()
+        return True
+    except OSError:
+        return False
+
+# Function to reallocate port if it is not available
+def reallocate_ports(env_name, port):
+    # Offset the original port by 10
+    new_port = int(port) + int(OFFSET)
+    while not is_port_open(str(new_port)):
+        new_port = int(port) + int(OFFSET)
+
+    # Set the new port value in the environment variable
+    os.environ[env_name] = str(new_port)
+    print("Port for {} changed to: {}".format(env_name, os.environ.get(env_name)))
+
+# Function to iterate over all ports and automatically reallocate
+def check_all_ports():
+    for x in PORTS:
+        if is_port_open(os.environ.get(x)):
+            continue
+        else:
+            reallocate_ports(x, os.environ.get(x))
+
+    print("All Preliminary Port Checks Completed")
+
+
 
 def main(processes, config_path, parachain_type):
+    ## Load environment file
+    load_dotenv('.env')
+    ## Check Ports and Automatically Reallocate
+    check_all_ports()
+
     print('Starting litentry-parachain in background')
 
     with open(config_path) as config_file:
         config = json.load(config_file)
+        config['workers'][0]['flags'] = [flag.replace('$CollatorWSPort', os.environ.get('CollatorWSPort', '')) for flag in config['workers'][0]['flags']]
```

Comment:
Looks like here only checks `worker0`. The config_file could have multiple worker configs.

Reply:
So I moved this to only happen when in `dev` mode wherein we will only have a single worker, Let me know what you think